integration/container/daemon_test.go

```diff
@@ -50,3 +51,50 @@ func TestContainerKillOnDaemonStart(t *testing.T) {
 	assert.Check(t, is.Nil(err))
 	assert.Assert(t, !inspect.State.Running)
 }
+
+// When the daemon doesn't stop in a clean way (eg. it crashes, the host has a power failure, etc..), or if it's started
+// with live-restore enabled, stopped containers should have their NetworkSettings cleaned up the next time the daemon
+// starts.
+func TestNetworkStateCleanupOnDaemonStart(t *testing.T) {
+	skip.If(t, testEnv.IsRemoteDaemon, "cannot start daemon on remote test run")
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
+	skip.If(t, testEnv.IsRootless, "scenario doesn't work with rootless mode")
+
+	t.Parallel()
+
+	ctx := testutil.StartSpan(baseContext, t)
+
+	d := daemon.New(t)
+	defer d.Cleanup(t)
+
+	d.StartWithBusybox(ctx, t)
+	defer d.Stop(t)
+
+	apiClient := d.NewClientT(t)
+
+	// The intention of this container is to ignore stop signals.
+	// Sadly this means the test will take longer, but at least this test can be parallelized.
+	cid := container.Run(ctx, t, apiClient,
+		container.WithExposedPorts("80/tcp"),
+		container.WithPortMap(nat.PortMap{"80/tcp": {{HostPort: "8080"}}}),
+		container.WithCmd("/bin/sh", "-c", "while true; do echo hello; sleep 1; done"))
+	defer func() {
+		err := apiClient.ContainerRemove(ctx, cid, containertypes.RemoveOptions{Force: true})
+		assert.NilError(t, err)
+	}()
+
+	inspect, err := apiClient.ContainerInspect(ctx, cid)
+	assert.NilError(t, err)
+	assert.Assert(t, inspect.NetworkSettings.SandboxID != "")
+	assert.Assert(t, inspect.NetworkSettings.SandboxKey != "")
+	assert.Assert(t, inspect.NetworkSettings.Ports["80/tcp"] != nil)
+
+	assert.NilError(t, d.Kill())
+	d.Start(t)
```

Comment:
I'm not sure if this is really an issue - but `Kill` only sends the signal, it doesn't hang around for the process to disappear. So, I wonder if there's a possible race between the OS cleaning things up and the `d.Start()`?